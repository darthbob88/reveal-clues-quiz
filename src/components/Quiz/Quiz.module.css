.question {
  border: 1px solid #bbb;
  margin-top: 1rem;
  padding: 0.5rem;
}
.disabled {
  background: #bcbcbc;
  color: #222;
}
.quiz {
  border: 1px solid lightgray;
  padding-bottom: 1em;
}
.startQuiz {
  opacity: 100%;
  display: inline-block;
  margin: 2rem 0.5rem 4rem;
}

div.nextPrevQs {
  display: flex;
  justify-content: center;
}
div.nextPrevQs button {
  margin: 0 1rem;
}
.questions li {
  display: inline-block;
  padding: 0.3rem;
  margin: 0.2rem;
  cursor: pointer;
}

.unanswered {
  background-color: #e6e6fa4f;
}
.correct {
  background-color: #7cfc007a;
}
.incorrect {
  background-color: #ff0000a6;
  color: white;
}

.current {
  border: 1px solid black;
}
