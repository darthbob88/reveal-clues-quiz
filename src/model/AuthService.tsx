import { firebaseAuth } from "../firebase";
import React, { useEffect, useState } from "react";
import { GoogleAuthProvider, User, createUserWithEmailAndPassword, sendPasswordResetEmail, signInWithEmailAndPassword, signInWithPopup } from "firebase/auth";

const doCreateUserWithEmailAndPassword = (
  email: string,
  password: string,
  userName: string
) =>
  createUserWithEmailAndPassword(firebaseAuth, email, password).then(result => {
    if (result.user == null) return false;
  }).catch(function (error) {
    console.log(error);
  });

const doSignInWithEmailAndPassword = (email: string, password: string) =>
  signInWithEmailAndPassword(firebaseAuth, email, password);

const doSignOut = () => firebaseAuth.signOut();

const doPasswordReset = (email: string) =>
  sendPasswordResetEmail(firebaseAuth, email);

// const doPasswordUpdate = (password: string) => {
//   if (firebaseAuth.currentUser != null) {
//     firebaseAuth.currentUser.updatePassword(password);
//   } else {
//     //TODO: If not logged in, throw an error?
//   }
// };

const doSignInWithGoogleOauth = () => {
  var provider = new GoogleAuthProvider();
  firebaseAuth.useDeviceLanguage();
  signInWithPopup(firebaseAuth, provider).then((result) => {
    // The signed-in user info.
    const user = result.user;
    console.log(user.displayName);
    // IdP data available using getAdditionalUserInfo(result)
    // ...
  }).catch((error) => {

  });

}


const currentUser: (User | null) = firebaseAuth.currentUser;
export const AuthService = {
  currentUser,
  doPasswordReset,
  doCreateUserWithEmailAndPassword,
  // doPasswordUpdate,
  doSignInWithEmailAndPassword,
  doSignInWithGoogleOauth,
  doSignOut
};

export const AuthContext = React.createContext(AuthService);

type AuthProviderTypes = { children: React.ReactNode };
export const AuthProvider: React.FunctionComponent<AuthProviderTypes> = ({
  children
}) => {
  const [currentUser, setCurrentUser] = useState<User | null>(null);
  useEffect(() => {
    firebaseAuth.onAuthStateChanged(user => setCurrentUser(user));
  });
  return (
    <AuthContext.Provider value={{ ...AuthService, currentUser }}>
      {children}
    </AuthContext.Provider>
  );
};
