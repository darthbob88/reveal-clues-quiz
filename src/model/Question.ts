export type Question = {
    clues: string[];
    answer: string;
    revealOnAnswer?: string;
}
export enum QuestionState {
    UNANSWERED,
    CORRECTLY_ANSWERED,
    INCORRECTLY_ANSWERED
}