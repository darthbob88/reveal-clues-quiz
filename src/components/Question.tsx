import React, { Component, useState } from "react";
import { Question } from "../model/Question";
type QuestionProps = {
  question: Question;
};
export const QuestionComponent: React.FunctionComponent<QuestionProps> = ({
  question,
}) => {
  const [revealedClues, setRevealedClues] = useState(1);
  const [currentGuess, setCurrentGuess] = useState("");
  /* PLAN: 
        <prompt>
        <show another clue button => increments numbers of clues shown, decrements number of points.>
        <clue 1>
        <either black bar or clue
    */
  return (
    <div>
      {/* Now where the hell do I get the prompt? */}
      <button onClick={() => setRevealedClues(revealedClues + 1)}>
        Reveal Another Clue
      </button>
      {question.clues.map((clue, index) => (
        <ClueComponent clue={clue} revealed={index <= revealedClues} />
      ))}
      <input
        value={currentGuess}
        onChange={(event) => setCurrentGuess(event?.currentTarget.value)}
      />
    </div>
  );
};

type ClueProps = { clue: string; revealed: boolean };
const ClueComponent: React.FunctionComponent<ClueProps> = ({
  clue,
  revealed,
}) => {
  return <span>{revealed ? clue : "-----------"}</span>;
};
