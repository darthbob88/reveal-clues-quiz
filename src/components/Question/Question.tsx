import { observer } from "mobx-react";
import React, { useState, useEffect } from "react";
import { Question, QuestionEnum, QuestionState } from "../../model/Question";
import styles from "./Question.module.css";

type QuestionProps = {
  question: Question;
  state: QuestionState;
  awardPoints: Function;
};
export const QuestionComp: React.FunctionComponent<QuestionProps> = ({
  question,
  state,
  awardPoints,
}) => {
  const [currentGuess, setCurrentGuess] = useState("");

  useEffect(() => {
    return () => {
      setCurrentGuess("");
    };
  }, [question]);

  const submitAnswer = () => {
    const qState: Partial<QuestionState> = {
      revealedClues: state.revealedClues,
    };

    if (
      currentGuess.toLocaleLowerCase() === question.answer.toLocaleLowerCase()
    ) {
      const questionValue = question.clues.length - state.revealedClues;
      qState.score = questionValue;
      qState.state = QuestionEnum.CORRECTLY_ANSWERED;
    } else {
      qState.state = QuestionEnum.INCORRECTLY_ANSWERED;
      qState.score = 0;
    }
    awardPoints(qState);
  };
  const revealOnAnswer = () => {
    if (QuestionEnum.UNANSWERED === state.state) {
      return null;
    } else if (QuestionEnum.CORRECTLY_ANSWERED === state.state) {
      const questionValue = question.clues.length - state.revealedClues;
      return <span>Correct! You get {questionValue} points.</span>;
    } else {
      return (
        <span>
          Incorrect! You get 0 points. The correct answer is {question.answer}.
        </span>
      );
    }
  };
  const revealAnotherClue = () => {
    state.revealedClues = state.revealedClues + 1;
  };

  return (
    <div className={styles.question}>
      <span className="prompt">In what state will you find...</span>
      {/* Now where the hell do I get the prompt? */}
      <button
        onClick={() => revealAnotherClue()}
        disabled={
          state.state !== QuestionEnum.UNANSWERED ||
          state.revealedClues === question.clues.length - 1
        }
      >
        Reveal Another Clue
      </button>
      <span>{1 + state.revealedClues} clues revealed</span>
      <ul className={styles.clues}>
        {question.clues.map((clue, index) => (
          <ClueComponent
            key={index}
            clue={clue}
            revealed={index <= state.revealedClues}
          />
        ))}
      </ul>
      <label>
        Your Answer:
        <input
          disabled={state.state !== QuestionEnum.UNANSWERED}
          value={currentGuess}
          onKeyPress={(event) => (event.key === "Enter" ? submitAnswer() : "")}
          onChange={(event) => setCurrentGuess(event?.currentTarget.value)}
        />
      </label>
      <button
        onClick={(event) => {
          submitAnswer();
        }}
        disabled={state.state !== QuestionEnum.UNANSWERED}
      >
        Submit Guess
      </button>
      {revealOnAnswer()}
    </div>
  );
};

export const QuestionComponent = observer(QuestionComp);

type ClueProps = { clue: string; revealed: boolean };
const ClueComponent: React.FunctionComponent<ClueProps> = ({
  clue,
  revealed,
}) => {
  return <li>{revealed ? clue : "-----------"}</li>;
};
