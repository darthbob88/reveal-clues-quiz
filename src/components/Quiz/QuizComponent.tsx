import React, { useState } from "react";
import { QuizState } from "../../model/Quiz";
import styles from "./Quiz.module.css";
import { QuestionComponent } from "../Question/Question";
import { QuestionEnum, QuestionState } from "../../model/Question";
import { observer } from "mobx-react";

type QuizProps = {
  quiz: QuizState;
};
export const QuizComp: React.FunctionComponent<QuizProps> = ({ quiz }) => {
  const [currentQuestion, setCurrentQuestion] = useState(0);
  const questions = quiz.currentQuiz.questions;

  const prevQuestion = () => {
    // This is stupid, but I can't think of a much better method
    let prev = questions.length + currentQuestion - 1;
    while (
      quiz.quizState[prev % questions.length].state !== QuestionEnum.UNANSWERED
    ) {
      prev--;
    }
    setCurrentQuestion(prev % questions.length);
  };

  const nextQuestion = () => {
    let next = currentQuestion + 1;
    while (
      quiz.quizState[next % questions.length].state !== QuestionEnum.UNANSWERED
    ) {
      next++;
    }
    setCurrentQuestion(next % questions.length);
  };

  const scoreQuestion = (points: QuestionState) => {
    quiz.scoreQuestion(currentQuestion, points);
    setTimeout(() => {
      nextQuestion();
    }, 800);
  };

  return (
    <div className={styles.quiz}>
      <p>
        Time remaining: {quiz.display} <br />
        Current score: {quiz.scorePoints} pts {quiz.scorePercent}/
        {quiz.quizState.length} question(s) correct
        {quiz.startedQuiz ? "" : <button className={styles.startQuiz} onClick={quiz.startQuiz}>Start Quiz</button>}
      </p>
      <div className={`${!quiz.startedQuiz ? styles.disabled : ""}`}>
        <ul className={styles.questions}>
          {quiz.quizState.map((question, index) => (
            <li
              key={index}
              onClick={() => setCurrentQuestion(index)}
              className={`${index === currentQuestion ? styles.current : ""}
              ${
                //TODO: This is ugly, but I can't find a good way to make this a mapping.
                question.state === QuestionEnum.UNANSWERED
                  ? styles.unanswered
                  : question.state === QuestionEnum.CORRECTLY_ANSWERED
                    ? styles.correct
                    : styles.incorrect
                }`}
            >
              {index + 1}
            </li>
          ))}
        </ul>
        <button onClick={() => prevQuestion()}>&lt; Previous Question</button>
        <button onClick={() => nextQuestion()}>Next Question &gt;</button>
        <QuestionComponent
          question={questions[currentQuestion]}
          state={quiz.quizState[currentQuestion]}
          awardPoints={scoreQuestion}
        />
      </div>
    </div>
  );
};

export const QuizComponent = observer(QuizComp);
