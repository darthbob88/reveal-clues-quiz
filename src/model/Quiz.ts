import { action, computed, observable } from "mobx";
import { defaultQuestionState, Question, QuestionEnum, QuestionState } from "./Question";
/**
 * Overarching type for a quiz.
 * @field prompt: string A prompt for each question, to be used if there is no overruling one in the individual question
 * @field questions: Question[] A set of questions for each quiz.
 * @field title: An overarching title for the quiz. 
 */
export type Quiz = {
    questions: Question[];
    title: string;
    prompt: string;
}

export const defaultQuiz: Quiz = {
    title: "States by Attraction",
    prompt: "In what state will you find...",

    questions: [{
        clues: [
            "Ernest Hemingway’s Grave",
            "The Smurf Turf",
            "The Yellowstone Zone of Death",
            "The <State> Potato Museum"
        ],
        answer: "Idaho"
    },
    {
        clues: [
            "Weeki Wachee, the city of mermaids",
            "The Ernest Hemingway Home and Museum",
            "NASA Vehicle Assembly Building, a building so large it has its own weather",
            "The Southernmost Point in the Continental US"
        ],
        answer: "Florida"
    },
    {
        clues: [
            "World's Largest Collection of Smallest Versions of Largest Things",
            "Monument Rocks",
            "The Equality House, across the street from Westboro",
            "Geographic Center of the Contiguous United States"
        ],
        answer: "Kansas"
    },
    {
        clues: [
            "Mutter Museum of medical oddities",
            "Fallingwater, designed by Frank Lloyd Wright",
            "Gobbler’s Knob, home of Punxsutawney Phil",
            "The grave of Fred McFeely Rogers"
        ],
        answer: "Pennsylvania"
    },
    {
        clues: [
            "Slocum Mechanical Puzzle Collection",
            'Mentone Egg, A giant concrete egg marks this town as the "Egg Basket of the Midwest."',
            "Kokomantis",
            "Hoosier Hill"
        ],
        answer: "Indiana",
        revealOnAnswer: "https://www.atlasobscura.com/places/slocum-mechanical-puzzle-collection  https://www.atlasobscura.com/places/mentone-egg  https://www.atlasobscura.com/places/kokomantis-scrap-art-sculpture https://www.atlasobscura.com/places/hoosier-hill"
    },
    {
        clues: [
            "USPS Remote Encoding Facility",
            "Pando, the trembling giant",
            "The Spiral Jetty, a sculpture built onto a lake which is only visible during droughts",
            "Bonneville Salt Flats/Speedway"
        ],
        answer: "Utah"
    },
    {
        clues: [
            "The Very Large Array radiotelescope, outside of Socorro.",
            "Smokey Bear’s Grave",
            "Carlsbad Caverns",
            "The International UFO Museum and Research Center"
        ],
        answer: "New Mexico"
    },
    {
        clues: [
            "Vacuum Cleaner Museum",
            "Vista House at Crown Point",
            "Evergreen Aviation Museum, home of the Spruce Goose",
            "International Rose Test Garden, in the City of Roses"
        ],
        answer: "Oregon"
    },
    {
        clues: [
            "Rhea County Courthouse, site of the Scopes monkey trial",
            "Concrete Parthenon, built for the 1897 Centennial Exposition",
            "National Civil Rights Museum, established in the hotel where Dr Martin Luther King Jr was assassinated",
            "Jack Daniels Distillery"
        ],
        answer: "Tennessee"
    },
    {
        clues: [
            "Dr. Seuss House aka The Goose Creek Tower, a 185 foot tall log cabin",
            "The giant metal chicken statue, outside the town of Chicken",
            "St Michael’s Cathedral, first Russian Orthodox Church in North America",
            "Santa Claus House, in the town of North Pole"
        ],
        answer: "Alaska"
    },
    {
        clues: [
            "Ape Cave, longest continuous lava tube in the continental US",
            "Maryhill Museum and Stonehenge",
            "Brown and Haley factory store, manufacturers of Almond Roca",
            "Beacon Rock, one of the largest monoliths in the world"
        ],
        answer: "Washington"
    }]
}
export class QuizState {
    @observable currentQuiz: Quiz;
    // TODO: It'd be neat if I could make this part of the quiz itself
    @observable quizState: QuestionState[] = [];
    @observable timeRemaining: number = 0;
    @observable startedQuiz = false;


    constructor(chosenQuiz: Quiz) {
        this.currentQuiz = chosenQuiz;
        this.quizState = chosenQuiz.questions.map(question => ({ ...defaultQuestionState }));
        this.timeRemaining = 8 * 60 * 1000;
    }

    //TODO: Three methods, all called score*, is a little confusing.
    scoreQuestion(questionIndex: number, questionState: QuestionState) {
        if (this.quizState[questionIndex].state === QuestionEnum.UNANSWERED) {
            this.quizState[questionIndex] = questionState;
        }
    }

    @computed get isComplete() {
        const allQsAnswered = this.quizState.every(question => question.state !== QuestionEnum.UNANSWERED);
        const outOfTime = this.timeRemaining <= 0;
        return allQsAnswered || outOfTime;
    }

    @computed get scorePoints() {
        return this.quizState.reduce((acc, cur) => acc + cur.score, 0);
    }

    //TODO: fix this to return a string or smth, "right / total"
    @computed get scorePercent() {
        const numCorrect = this.quizState.filter(question => question.state === QuestionEnum.CORRECTLY_ANSWERED);
        return numCorrect.length;
    }

    startQuiz = () => {
        this.startedQuiz = true;
        this.measure();
    }

    @action measure() {
        if (!this.startedQuiz) return;

        this.timeRemaining -= 50;

        if (this.isComplete) {
            alert(`Time's up. Final score is ${this.scorePoints}`);
            return;
        }
        setTimeout(() => this.measure(), 50);
    }

    @computed get display() {
        const tenMilliSeconds = parseInt((this.timeRemaining / 10).toString(), 10);

        const seconds = parseInt((tenMilliSeconds / 100).toString(), 10);
        const minutes = parseInt((seconds / 60).toString(), 10);
        // TODO: Get some good method for formatting this for display.
        return `${minutes} : ${seconds % 60} :  ${tenMilliSeconds % 100}`;
    }
} 