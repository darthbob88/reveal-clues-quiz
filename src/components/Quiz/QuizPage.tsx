import React, { useEffect, useState } from "react";
import { useParams } from "react-router-dom";
import { Quiz, QuizState } from "../../model/Quiz";
import { loadQuiz } from "../../model/QuizService";
import { QuizComponent } from "./QuizComponent";

export const QuizPage: React.FunctionComponent = () => {
    const { id } = useParams<{ id: string }>();
    const [quiz, setQuiz] = useState<Quiz>();
    useEffect(() => {
        const fetchData = async (id: string) => {
            const result = await loadQuiz(id);
            setQuiz(result);
        };
        id != null && fetchData(id);
    }, [id]);
    return (
        quiz === undefined ? <h2>Still Loading</h2> : <QuizComponent quiz={new QuizState(quiz)} />
    );
}