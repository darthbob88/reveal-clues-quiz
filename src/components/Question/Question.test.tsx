import React from "react";
import { render, fireEvent } from "@testing-library/react";
import { QuestionComponent } from "./Question";
import { defaultQuiz, defaultQuizState } from "../../model/Quiz";
import { QuestionEnum, QuestionState } from "../../model/Question";

const defaultQuestion = defaultQuiz.questions[0];
const awardPoints = (score: number) => {
  console.log(score);
};
test("renders a question properly", () => {
  const { getByText, getAllByText } = render(
    <QuestionComponent
      state={defaultQuizState}
      question={defaultQuestion}
      awardPoints={awardPoints}
    />
  );
  const firstClue = getByText(defaultQuestion.clues[0]);
  expect(firstClue).toBeInTheDocument();
  const unrevealedClues = getAllByText("-----------");
  expect(unrevealedClues.length).toBe(defaultQuestion.clues.length - 1);

  const revealClue = getByText("Reveal Another Clue");
  expect(revealClue).toBeEnabled();
});
// Can't easily test this after moving `revealedClues` to MobX :(
xtest("reveals more clues as necessary", () => {
  const { queryByText, getByText, getAllByText } = render(
    <QuestionComponent
      state={defaultQuizState}
      question={defaultQuestion}
      awardPoints={awardPoints}
    />
  );
  const revealClue = getByText("Reveal Another Clue");
  expect(revealClue).toBeEnabled();

  // Should only show one clue at first
  const firstClue = getByText(defaultQuestion.clues[0]);
  expect(firstClue).toBeInTheDocument();
  let unrevealedClues = getAllByText("-----------");
  expect(unrevealedClues.length).toBe(defaultQuestion.clues.length - 1);

  // Should reveal a second clue
  fireEvent.click(revealClue);
  const secondClue = getByText(defaultQuestion.clues[1]);
  expect(secondClue).toBeInTheDocument();
  unrevealedClues = getAllByText("-----------");
  expect(unrevealedClues.length).toBe(defaultQuestion.clues.length - 2);

  // Should reveal a third clue
  fireEvent.click(revealClue);
  const thirdClue = getByText(defaultQuestion.clues[2]);
  expect(thirdClue).toBeInTheDocument();
  unrevealedClues = getAllByText("-----------");
  expect(unrevealedClues.length).toBe(defaultQuestion.clues.length - 3);

  // Should reveal all four clues
  fireEvent.click(revealClue);
  const fourthClue = getByText(defaultQuestion.clues[3]);
  expect(fourthClue).toBeInTheDocument();
  expect(queryByText("-----------")).not.toBeInTheDocument();

  // After we reveal all four clues, the "Reveal Clue" button should be disabled
  expect(revealClue).toBeDisabled();
});

test("awards 4 points for correct answer with 1 clue", () => {
  const incrementScore = jest.fn();
  const { getByText, getByLabelText, queryByText } = render(
    <QuestionComponent
      state={defaultQuizState}
      question={defaultQuestion}
      awardPoints={incrementScore}
    />
  );
  const answerSlot = getByLabelText(/Answer/i);
  fireEvent.change(answerSlot, { target: { value: defaultQuestion.answer } });

  const submitBtn = getByText(/Submit/i);
  fireEvent.click(submitBtn);

  const expectedState: QuestionState = {
    state: QuestionEnum.CORRECTLY_ANSWERED,
    score: defaultQuestion.clues.length,
    revealedClues: 0,
  };
  expect(incrementScore).toHaveBeenCalledTimes(1);
  expect(incrementScore).toHaveBeenCalledWith(expectedState);
});

test("awards 3 points for correct answer with 2 clues", () => {
  const incrementScore = jest.fn();
  const { getByText, getByLabelText } = render(
    <QuestionComponent
      state={defaultQuizState}
      question={defaultQuestion}
      awardPoints={incrementScore}
    />
  );

  const revealClue = getByText("Reveal Another Clue");
  expect(revealClue).toBeEnabled();
  fireEvent.click(revealClue);

  const answerSlot = getByLabelText(/Answer/i);
  fireEvent.change(answerSlot, { target: { value: defaultQuestion.answer } });

  const submitBtn = getByText(/Submit/i);
  fireEvent.click(submitBtn);

  const expectedState: QuestionState = {
    state: QuestionEnum.CORRECTLY_ANSWERED,
    score: defaultQuestion.clues.length - 1,
    revealedClues: 1,
  };
  expect(incrementScore).toHaveBeenCalledTimes(1);
  expect(incrementScore).toHaveBeenCalledWith(expectedState);
});

test("awards 0 points for incorrect answer", () => {
  const incrementScore = jest.fn();
  const { getByText, getByLabelText } = render(
    <QuestionComponent
      state={defaultQuizState}
      question={defaultQuestion}
      awardPoints={incrementScore}
    />
  );
  const answerSlot = getByLabelText(/Answer/i);
  fireEvent.change(answerSlot, { target: { value: "butts" } });

  const submitBtn = getByText(/Submit/i);
  fireEvent.click(submitBtn);

  const expectedState: QuestionState = {
    state: QuestionEnum.INCORRECTLY_ANSWERED,
    score: 0,
    revealedClues: 1,
  };
  expect(incrementScore).toHaveBeenCalledTimes(1);
  expect(incrementScore).toHaveBeenCalledWith(expectedState);
});
