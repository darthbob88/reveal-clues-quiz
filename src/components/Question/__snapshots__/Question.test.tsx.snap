// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a question properly 1`] = `
<div>
  <div
    class="question"
  >
    <span
      class="prompt"
    >
      In what state will you find...
    </span>
    <button>
      Reveal Another Clue
    </button>
    <span>
      1
       clues revealed
    </span>
    <ul
      class="clues"
    >
      <li>
        Ernest Hemingway's Grave
      </li>
      <li>
        -----------
      </li>
      <li>
        -----------
      </li>
      <li>
        -----------
      </li>
    </ul>
    <label>
      Your Answer:
      <input
        value=""
      />
    </label>
    <button>
      Submit Guess
    </button>
  </div>
</div>
`;
